embassy-rp/src/pwm.rs

```diff
@@ -344,6 +344,62 @@ impl<'d> Pwm<'d> {
     fn bit(&self) -> u32 {
         1 << self.slice as usize
     }
+
+    #[inline]
+    /// Split Pwm driver to allow separate duty cycle control of each channel
+    pub fn split(&self) -> (PwmOutput,PwmOutput){
```

Comment:
we need to ensure only one instance of the pwm channel exists. If this takes `&self` you can call it multiple times and obtain multiple copies of `PwmOutput`.

I suggest adding both an owned `split` that permanently consumes `self`, and a "temporary" split that only borrows it, but mutably so you can't call it again until you drop the `PwmOutput`s.

```rust
pub fn split(self) -> (PwmOutput<'d>, PwmOutput<'d>) {...}
pub fn split_by_ref(&mut self) -> (PwmOutput<'_>, PwmOutput<'_>) {...}
```